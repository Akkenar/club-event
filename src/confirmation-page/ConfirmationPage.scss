@import '../style/placeholders';
@import url('../../node_modules/semantic-ui-css/components/button.min.css');

.ConfirmationPage__print {
  @extend %no-print;
  @extend %full-width;

  display: block !important;
  margin: 0 auto !important;
}
