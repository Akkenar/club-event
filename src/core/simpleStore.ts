/*
 * Simple store. No need to bother with Redux/Flux with such a simple usecase.
 * */

import { Registration } from '../register-page/register.type';

let registration: Registration | null = null;

export function setSimpleStore(newRegistration: Registration) {
  registration = newRegistration ? { ...newRegistration } : null;
}

export function getSimpleStore(): Registration | null {
  return registration;
}
