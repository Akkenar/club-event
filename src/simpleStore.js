/*
 * Simple store. No need to bother with Redux/Flux with such a simple usecase.
 * */

let simpleState = {};

export function setSimpleStore(newState) {
  simpleState = Object.assign({}, simpleState, newState);
}

export function getSimpleStore() {
  return simpleState;
}
