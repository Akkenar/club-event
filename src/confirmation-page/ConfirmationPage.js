import React, { Fragment, useContext, useEffect } from 'react';
import getKey from '../core/intl/getKey';
import LanguageContext from '../core/intl/LanguageContext';
import { setPageTitle } from '../page.lib';
import { Button, Header } from 'semantic-ui-react';

import { getSimpleStore } from '../simpleStore';
import LazyImage from '../core/lazy-image/LazyImage';

import img from '../assets/Zsolt-Sarkozi-Grottes-aux-Fees-17.jpg';
import imgwebp from '../assets/Zsolt-Sarkozi-Grottes-aux-Fees-17.webp';
import OrderRecap from './order-recap/OrderRecap';

import './ConfirmationPage.scss';

function format(confirmation, total, reference) {
  if (!confirmation) {
    return confirmation;
  }

  return confirmation
    .replace('%TOTAL%', total)
    .replace('%REFERENCE%', reference);
}

function printPage() {
  window.print();
}

function supportsPrint() {
  return window.print === 'function';
}

const ConfirmationPage = () => {
  const { messages } = useContext(LanguageContext);
  useEffect(() => {
    setPageTitle(getKey('confirmation.page.title', messages));
  });

  // From the register form.
  const data = getSimpleStore();
  const { total, reference } = data;
  const confirmationMessage = format(messages.confirmation, total, reference);

  const printButton = supportsPrint() ? (
    <Button
      color="green"
      onClick={printPage}
      className="ConfirmationPage__print"
    >
      {getKey('confirmation.page.print', messages)}
    </Button>
  ) : null;

  return (
    <Fragment>
      <Header as="h1">{getKey('confirmation.page.title', messages)}</Header>
      <div dangerouslySetInnerHTML={{ __html: confirmationMessage }} />
      <Header as="h2">
        {getKey('confirmation.page.recap.title', messages)}
      </Header>
      <OrderRecap {...data} />
      {printButton}
      <LazyImage
        className="no-print"
        width={320}
        src={img}
        srcwebp={imgwebp}
        alt={'Photo: Zsolt Sarkozi'}
      />
    </Fragment>
  );
};

export default ConfirmationPage;
