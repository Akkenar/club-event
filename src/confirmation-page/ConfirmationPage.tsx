import React from 'react';
import { useContext, useEffect } from 'react';
import { Button, Header } from 'semantic-ui-react';
import * as img from '../assets/Zsolt-Sarkozi-Grottes-aux-Fees-17.jpg';
import getKey from '../core/intl/getKey';
import LanguageContext from '../core/intl/LanguageContext';
import LazyImage from '../core/lazy-image/LazyImage';
import OrderRecap from '../core/order-recap/OrderRecap';
import { setPageTitle } from '../core/page.lib';
import { getSimpleStore } from '../core/simpleStore';

import './ConfirmationPage.scss';

function format(
  confirmation: string,
  total: number | null,
  reference: string,
): string {
  if (!confirmation) {
    return confirmation;
  }

  return confirmation
    .replace('%TOTAL%', (total || '').toString())
    .replace('%REFERENCE%', reference);
}

function printPage() {
  window.print();
}

function supportsPrint() {
  return typeof window.print === 'function';
}

const ConfirmationPage = () => {
  const { messages, confirmation } = useContext(LanguageContext);
  useEffect(() => {
    setPageTitle(getKey('confirmation.page.title', messages));
  });

  // From the register form.
  const registration = getSimpleStore();
  const { total = null, reference = '' } = registration || {};
  const confirmationMessage = format(confirmation, total, reference);

  return (
    <div className="ConfirmationPage">
      <Header as="h1" data-testid="main-title">
        {getKey('confirmation.page.title', messages)}
      </Header>
      <div dangerouslySetInnerHTML={{ __html: confirmationMessage }} />
      <Header as="h2">
        {getKey('confirmation.page.recap.title', messages)}
      </Header>
      {registration ? <OrderRecap registration={registration} /> : null}
      <Button
        disabled={!supportsPrint()}
        color="green"
        onClick={printPage}
        className="ConfirmationPage__print"
      >
        {getKey('confirmation.page.print', messages)}
      </Button>
      <LazyImage
        className="no-print"
        width={320}
        height={400}
        src={img}
        alt={'Photo: Zsolt Sarkozi'}
      />
    </div>
  );
};

export default ConfirmationPage;
