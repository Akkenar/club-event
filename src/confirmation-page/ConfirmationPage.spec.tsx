import * as React from 'react';
import { render } from 'react-testing-library';
import { setSimpleStore } from '../simpleStore';
import ConfirmationPage from './ConfirmationPage';

describe('ConfirmationPage', () => {
  it('should match the snapshot', () => {
    setSimpleStore({
      products: {
        breakfast: 1,
        camping: 1,
        dinner: 4,
        itemSize1: 2,
        itemSize2: 2,
        itemSize3: 2,
        itemSize4: 2,
        picknick: 1,
        sleeping: 1,
        vegetarian: 4,
      },
    });
    const wrapper = render(<ConfirmationPage />);
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should set the page title', () => {
    // Will be set by side effect through a hook
    render(<ConfirmationPage />);
    expect(document.title).toEqual('confirmation.page.title');
  });
});
