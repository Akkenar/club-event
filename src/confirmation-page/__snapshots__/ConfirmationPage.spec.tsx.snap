// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationPage should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ConfirmationPage"
    >
      <h1
        class="ui header"
      >
        confirmation.page.title
      </h1>
      <div />
      <h2
        class="ui header"
      >
        confirmation.page.recap.title
      </h2>
      <div
        class="ui raised segment OrderRecap"
      >
        <dl
          class="OrderRecap__container"
        >
          <dd
            class="OrderRecap__title"
          >
            register.form.firstName
          </dd>
          <dt
            class="OrderRecap__value"
          />
          <dd
            class="OrderRecap__title"
          >
            register.form.lastName
          </dd>
          <dt
            class="OrderRecap__value"
          />
          <dd
            class="OrderRecap__title"
          >
            register.form.club
          </dd>
          <dt
            class="OrderRecap__value"
          />
          <dd
            class="OrderRecap__title"
          >
            register.form.address
          </dd>
          <dt
            class="OrderRecap__value"
          >
            <address>
              <div>
                 
              </div>
              <div>
                 
              </div>
            </address>
          </dt>
          <dd
            class="OrderRecap__title"
          >
            register.form.counts.total
          </dd>
          <dt
            class="OrderRecap__value"
          >
            CHF 
          </dt>
        </dl>
        <dl
          class="OrderRecap__container"
        >
          <dd
            class="OrderRecap__title"
          >
            register.form.counts.breakfast
          </dd>
          <dt
            class="OrderRecap__value"
          >
            1
          </dt>
          <dd
            class="OrderRecap__title"
          >
            register.form.counts.camping
          </dd>
          <dt
            class="OrderRecap__value"
          >
            1
          </dt>
          <dd
            class="OrderRecap__title"
          >
            register.form.counts.dinner
          </dd>
          <dt
            class="OrderRecap__value"
          >
            4
          </dt>
          <dd
            class="OrderRecap__title"
          >
            register.form.counts.itemSize1
          </dd>
          <dt
            class="OrderRecap__value"
          >
            2
          </dt>
          <dd
            class="OrderRecap__title"
          >
            register.form.counts.itemSize2
          </dd>
          <dt
            class="OrderRecap__value"
          >
            2
          </dt>
          <dd
            class="OrderRecap__title"
          >
            register.form.counts.itemSize3
          </dd>
          <dt
            class="OrderRecap__value"
          >
            2
          </dt>
          <dd
            class="OrderRecap__title"
          >
            register.form.counts.itemSize4
          </dd>
          <dt
            class="OrderRecap__value"
          >
            2
          </dt>
          <dd
            class="OrderRecap__title"
          >
            register.form.counts.picknick
          </dd>
          <dt
            class="OrderRecap__value"
          >
            1
          </dt>
          <dd
            class="OrderRecap__title"
          >
            register.form.counts.sleeping
          </dd>
          <dt
            class="OrderRecap__value"
          >
            1
          </dt>
          <dd
            class="OrderRecap__title"
          >
            register.form.counts.vegetarian
          </dd>
          <dt
            class="OrderRecap__value"
          >
            4
          </dt>
        </dl>
      </div>
      <button
        class="ui green button ConfirmationPage__print"
      >
        confirmation.page.print
      </button>
      <svg
        aria-label="Photo: Zsolt Sarkozi"
        class="LazyImage no-print"
        height="400"
        viewBox="0 0 1024 1024"
        width="320"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d=""
        />
      </svg>
    </div>
  </div>
</body>
`;
